semantic-core/src/Core/Parser.hs

```diff
@@ -61,7 +61,12 @@ application :: (TokenParsing m, Carrier sig t, Member Core sig, Monad m) => m (t
 application = projection `chainl1` (pure (Core.$$))
 
 projection :: (TokenParsing m, Carrier sig t, Member Core sig, Monad m) => m (t Name)
-projection = foldl' (Core....) <$> atom <*> many (namedValue <$ dot <*> name)
+projection = let a <$$> b = flip a <$> b in do
+  head <- atom
+  res <- many (choice [ (Core..?)  <$$> (symbol ".?" *> identifier)
+                      , (Core....) <$$> (dot *> identifier)
```

Comment:
I think these could be rephrased more idiomatically as:

```haskell
[ flip (Core..?)  <$ symbol ".?" <*> identifier
, flip (Core....) <$ dot <*> identifier
]
```

Reply:
You’re right, that is nicer.